pkg/manager/member/executor_manager.go

```diff
@@ -650,36 +650,29 @@ func (m *executorMemberManager) syncExecutorStatus(tc *v1alpha1.TiflowCluster, s
 		return err
 	}
 
-	if tc.ExecutorStsDesiredReplicas() == *sts.Spec.Replicas {
+	if tc.ExecutorStsDesiredReplicas() != *sts.Spec.Replicas {
 		tc.Status.Executor.Phase = v1alpha1.ScalePhase
 	} else if upgrading {
 		tc.Status.Executor.Phase = v1alpha1.UpgradePhase
 	} else {
 		tc.Status.Executor.Phase = v1alpha1.NormalPhase
 	}
 
-	// todo: Get information about the Executor Members, FailureMembers and FailoverUID through the Master API
-	// todo: Or may be get info through the Sts Status
-	// TOBE
-	tiflowClient := tiflowapi.GetMasterClient(m.cli, ns, tcName, "", tc.IsClusterTLSEnabled())
-	_, err = tiflowClient.GetLeader()
+	tc.Status.Executor.Members, err = m.syncExecutorMembersStatus(tc)
 	if err != nil {
-		tc.Status.Master.Synced = false
 		return err
 	}
-
-	// todo: get member's infos for master
-	// todo: get failure members and UID
+	tc.Status.Executor.Synced = true
 
 	// get follows from podName
 	tc.Status.Executor.Image = ""
-	if c := m.getContainerByName(sts, "tiflow-executor"); c != nil {
+	if c := m.findContainerByName(sts, "tiflow-executor"); c != nil {
 		tc.Status.Executor.Image = c.Image
 	}
 
 	// todo: Need to get the info of volumes which running container has bound
 	// todo: Waiting for discussion
-	// vols,err := m.getVolsByName()
+	//m.syncVolsStatus(tc,sts)
 	tc.Status.Executor.Volumes = nil
```

Comment:
just a question, i can't find the code about `call client to update tc ` can you tell me where it is?

Reply:
In executor_manager.go  `L761` and `L798`, we do that